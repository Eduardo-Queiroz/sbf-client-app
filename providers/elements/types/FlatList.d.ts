declare type FlatListProps = import('react-native').FlatListProps<any>;
