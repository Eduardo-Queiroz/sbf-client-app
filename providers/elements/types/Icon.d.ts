interface SizeProps {
  size?: number;
  width?: number;
  height?: number;
}

interface IconPropsInterface extends SizeProps {
  height?: number;
  width?: number;
  size?: number;
  color?: import('@sbf-types/general').Color;
  icon: React.FC<any> | import('@sbf-providers/icons').IconsType;
}

declare type IconProps = IconPropsInterface;
