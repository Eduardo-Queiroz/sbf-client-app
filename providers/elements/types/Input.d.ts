declare type InputProps = import('react-native').TextInputBase &
  TrackeablePropsInterface;
