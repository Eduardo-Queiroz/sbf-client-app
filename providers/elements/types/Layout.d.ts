declare interface LayoutProps {
  flex?: number;
  style?: {};
  uiDebug?: boolean;
  width?: string;
  height?: string;
  round?: boolean;
  circle?: boolean;
  shaped?: boolean;
  marginTop?: import('@sbf-types/general').Gutter;
  elevated?: boolean;
  marginLeft?: import('@sbf-types/general').Gutter;
  paddingTop?: import('@sbf-types/general').Gutter;
  borderColor?: import('@sbf-types/general').Color;
  paddingLeft?: import('@sbf-types/general').Gutter;
  borderWidth?: string;
  bordered?: boolean;
  gutterOuter?: import('@sbf-types/general').Gutter;
  gutterInner?: import('@sbf-types/general').Gutter;
  gutterHorizontal?: import('@sbf-types/general').Gutter;
  gutterVertical?: import('@sbf-types/general').Gutter;
  marginRight?: import('@sbf-types/general').Gutter;
  textAlign?: import('@sbf-types/general').TextAlign;
  paddingRight?: import('@sbf-types/general').Gutter;
  marginBottom?: import('@sbf-types/general').Gutter;
  paddingBottom?: import('@sbf-types/general').Gutter;
  backgroundColor?: import('@sbf-types/general').Color;
  overflow?: 'visible' | 'hidden';
  fullWidth?: boolean;
  fullHeight?: boolean;
  alignItems?: import('@sbf-types/general').FlexAlign;
  alignContent?: import('@sbf-types/general').FlexAlign;
  justifyContent?: import('@sbf-types/general').FlexJustify;
  flexDirection?: import('@sbf-types/general').FlexDirections;
}

declare interface GridProps extends LayoutProps {
  wrap?: import('@sbf-types/general').FlexWrap;
  flex?: string | number;
}

declare interface GridItemProps extends LayoutProps {
  wrap?: import('@sbf-types/general').FlexWrap;
  flex?: string | number;
  col?: number;
  offset?: number;
  flexBasis?: string;
}

declare interface PositionProps extends LayoutProps {
  top?: string;
  left?: string;
  right?: string;
  width?: string;
  bottom?: string;
  zIndex?: number;
  position?: import('@sbf-types/general').Position;
}
