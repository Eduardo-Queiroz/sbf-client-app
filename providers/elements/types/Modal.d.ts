interface OwnProps {
  onDismiss?: () => void;
  onConfirm?: (value?: string) => void;
}

declare type ModalProps = import('react-native').ModalProps &
  OwnProps &
  TrackeablePropsInterface;
