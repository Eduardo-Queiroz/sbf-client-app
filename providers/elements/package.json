{
  "name": "@sbf-providers/elements",
  "main": "./src/index.tsx",
  "version": "1.0.0",
  "author": "Eduardo Queiroz",
  "license": "ISC"
}
