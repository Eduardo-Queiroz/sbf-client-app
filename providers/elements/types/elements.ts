export enum ElementsType {
  FLAT_LIST = 'FLAT_LIST',
  FORM = 'FORM',
  ICON = 'ICON',
  LAYOUT_ITEM = 'LAYOUT_ITEM',
  GRID = 'GRID',
  GRID_ITEM = 'GRID_ITEM',
  SAFE_AREA_LAYOUT_ITEM = 'SAFE_AREA_LAYOUT_ITEM',
  SAFE_AREA_GRID_ITEM = 'SAFE_AREA_GRID_ITEM',
  SCROLL_LAYOUT_ITEM = 'SCROLL_LAYOUT_ITEM',
  TOUCHABLE_CONTAINER = 'TOUCHABLE_CONTAINER',
  LOADING = 'LOADING',
  POSITION_ELEMENT = 'POSITION_ELEMENT',
  TOUCHABLE = 'TOUCHABLE',
  TYPOGRAPHY = 'TYPOGRAPHY',
  INPUT = 'INPUT',
  TRACKEABLE_HOC = 'TRACKEABLE_HOC',
  PICKER = 'PICKER',
  MODAL = 'MODAL',
  RATING = 'RATING',
}
