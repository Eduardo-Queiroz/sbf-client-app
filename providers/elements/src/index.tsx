import React, {PureComponent} from 'react';
import {ElementsType} from '../types/elements';
export {ElementsType} from '../types/elements';

export type Element = {[key in ElementsType]: React.FC<any>};

interface ElementsContext {
  elements?: Element;
}

export const context = React.createContext<ElementsContext>({});

export class ElementProvider extends PureComponent<ElementsContext> {
  public render() {
    const {elements, children} = this.props;
    return (
      <context.Provider
        value={{
          elements,
        }}>
        {children}
      </context.Provider>
    );
  }
}
export const useElement = <PropsType extends unknown>(
  type: ElementsType,
): React.FC<PropsType> => {
  const {elements} = React.useContext<ElementsContext>(context);
  if (!elements) throw new Error(`Providers de elementos n√£o inicializados`);
  return elements[type] || <></>;
};
