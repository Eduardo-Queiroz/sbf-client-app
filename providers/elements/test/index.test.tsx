/* eslint-disable react/display-name */
import React from 'react';
import {ElementProvider, useElement, context} from './../src/index';
import {ElementsType} from './../types/elements';
import {render} from '@testing-library/react-native';
import {renderHook} from '@testing-library/react-hooks';

const makeWrapper = ({children}: {children?: any}) => (
  <context.Provider value={{elements: {[ElementsType.FLAT_LIST]: true}}}>
    {children}
  </context.Provider>
);

describe('it should test elements provider', () => {
  test('it should verify provider snapshot', () => {
    const {toJSON} = render(
      <ElementProvider elements={{[ElementsType.FLAT_LIST]: true}} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('it should verify empty provider context', () => {
    try {
      const {result} = renderHook(() => useElement(ElementsType.FLAT_LIST));
      expect(result).toBe(false);
    } catch (e) {
      expect(e.message).toBe('Providers de elementos nÃ£o inicializados');
    }
  });

  test('it should verify specific value from provider context', async () => {
    const {result} = renderHook(() => useElement(ElementsType.FLAT_LIST), {
      wrapper: makeWrapper,
    });
    expect(result.current).toBe(true);
  });
});
