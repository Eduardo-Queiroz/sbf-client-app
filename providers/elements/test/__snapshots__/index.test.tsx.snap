// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should test elements provider it should verify provider snapshot 1`] = `null`;
